pkg/registry/core/pod/strategy.go

```diff
@@ -668,6 +669,56 @@ func validateContainer(container string, pod *api.Pod) (string, error) {
 	return container, nil
 }
 
+// applyLabelKeysToLabelSelector obtains the label value from the given label set by the key in labelKeys,
+// and merge to LabelSelector with the given operator:
+func applyLabelKeysToLabelSelector(labelSelector *metav1.LabelSelector, labelKeys []string, operator metav1.LabelSelectorOperator, podLabels map[string]string) {
+	for _, key := range labelKeys {
+		if value, ok := podLabels[key]; ok {
+			labelSelector.MatchExpressions = append(labelSelector.MatchExpressions, metav1.LabelSelectorRequirement{
```

Comment:
Yes, correct. Only kube-apiserver looks at this new field and append labelselector so that the scheduler indirectly achieves this matchLabelKeys feature.